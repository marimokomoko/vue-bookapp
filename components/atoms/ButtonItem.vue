<template>
  <v-row class="button">
    <v-col>
      <v-btn min-width="110" :to="url" :color="color">
        {{ buttonText }}
      </v-btn>
    </v-col>
  </v-row>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
@Component({})
export default class ButtonItem extends Vue {
  @Prop({ type: String, required: true })
  buttonText!: string

  @Prop({ type: String, required: true })
  url!: string

  @Prop({ type: String, required: true })
  color!: string
}
</script>
