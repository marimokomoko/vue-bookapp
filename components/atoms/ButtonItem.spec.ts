import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
import Vuetify from 'vuetify'
import ButtonItem from '@/components/atoms/ButtonItem.vue'

let wrapper: Wrapper<Vue>;

describe('画面レイアウト', () => {
    it('スナップショットテスト', () => {
        wrapper = mount(ButtonItem, {
            propsData: {
                buttonText: 'test',
                url: 'test',
                color: 'primary'
            },
            localVue: createLocalVue(),
            vuetify: new Vuetify(),
            stubs: ['router-link']
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})