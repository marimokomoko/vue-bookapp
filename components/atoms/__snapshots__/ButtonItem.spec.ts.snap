// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`画面レイアウト スナップショットテスト 1`] = `
"<div class=\\"row button\\">
  <div class=\\"col\\">
    <router-link-stub class=\\"v-btn v-btn--is-elevated v-btn--has-bg v-btn--router theme--light v-size--default primary\\" style=\\"min-width: 110px;\\"><span class=\\"v-btn__content\\">
      test
    </span></router-link-stub>
  </div>
</div>"
`;
